### PR TITLE
Added functions to utilities file.

### PR DESCRIPTION
**Pull request by Injector-Spenral**
Added:
get_vertical_angle()
get_horizontal_angle()
get_distance()
scale_value()

### CODE REVIEW

File: utilities/functions.py [hunk 1]
```diff
@@ -1,7 +1,40 @@
...
+import math
 
 
+def scale_value(
+    value: float,
+    input_lower: float,
+    input_upper: float,
+    output_lower: float,
+    output_upper: float,
+    exponent: float = 1,
```
Comment (raulvera59) [inline utilities/functions.py:12]: Why does this function also raise to a power? That is a slow operation, that won't be needed most of the time, and it's easy enough to apply to the result. I would remove it.

File: utilities/functions.py [hunk 2]
```diff
@@ -53,3 +86,69 @@ def get_corners_from_contour(contour: np.ndarray, corner_number=4) -> None:
             upper = current
         i += 1
     return hull
+
+
+# The next two functions' `x` and `y` arguments are in camera coordinates.
+# The origin is the top left corner of the screen, with left and down being positive.
```
Comment (raulvera59) [inline utilities/functions.py]: left is positive? If 0,0 is at the top-left corner, then right is positive. Either this comment is wrong, or the code is.

File: utilities/functions.py [hunk 3]
```diff
@@ -53,3 +86,69 @@ def get_corners_from_contour(contour: np.ndarray, corner_number=4) -> None:
...
+
+# Both have an `invert` argument which, if False, leaves the return angle with negative
```
Comment (raulvera59) [inline utilities/functions.py:97]: "negative at the origin" and "positive at the origin" are oxymorons. They are 0 at the origin, by definition. This comment does not explain what you mean.

File: utilities/functions.py [hunk 4]
```diff
@@ -53,3 +86,69 @@ def get_corners_from_contour(contour: np.ndarray, corner_number=4) -> None:
...
+    """Gets the ground distance from the camera to the target.
+
+    Args:
+        target_angle: The angle of the point below the centre plane of the camera. 
+            Downwards is positive, as returned by get_vertical_angle()'s default.
+        target_height: The height of the target above the ground (in metres)
+        camera_height: The height of the camera above the ground (in metres)
+        camera_tilt: The camera's angle of elevation from the ground (upwards is positive)
+    Returns:
+        A positive perpendicular distance to the target along the ground (in metres)
```
Comment (raulvera59) [inline utilities/functions.py]: "perpendicular"? I would remove this word. It's just the distance to the target along the ground.

**Review by raulvera59 — APPROVED** [review]
I'll approve, but I don't like the exponent, and some of the comments aren't clear.

File: utilities/functions.py (hunk 1 above)
Comment (Injector-Spenral) [reply utilities/functions.py:12]: I've made a check for whether or not it should do the exponent to hopefully save some time. The reason for this is that in past we have wanted to have an exponent built in to the function.

File: utilities/functions.py (hunk 2 above)
Comment (Injector-Spenral) [reply utilities/functions.py]: You're right, I've changed that.

File: utilities/functions.py (hunk 3 above)
Comment (Injector-Spenral) [reply utilities/functions.py:97]: I have clarified this.

File: utilities/functions.py (hunk 4 above)
Comment (Injector-Spenral) [reply utilities/functions.py]: Removed.

**Pull request merged**